### PR TITLE
Skip rebind of services after testing phase

### PR DESCRIPTION
**Pull request by radito3**
LMCROSSITXSADEPLOY-1868

### CODE REVIEW

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/listeners/EnteredTestingPhaseListener.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+@Named("enteredTestingPhaseListener")
+public class EnteredTestingPhaseListener extends AbstractProcessExecutionListener {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EnteredTestingPhaseListener.class);
+
+    @Override
+    protected void notifyInternal(DelegateExecution context)  {
+        context.setVariable(Constants.VAR_ENTERED_TESTING_PHASE, true);
+    }
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/listeners/EnteredTestingPhaseListener.java]: See `PrepareAppsRestartStep`. Can't we do something similar to what is done there?

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/Constants.java [hunk 2]
```diff
@@ -166,6 +166,7 @@
     public static final String VAR_HOOK_FOR_EXECUTION = "hookForExecution";
     public static final String VAR_FILE_ENTRIES = "fileEntries";
     public static final String VAR_APPS_TO_RENAME = "appsToRename";
+    public static final String VAR_PASSED_TESTING_PHASE = "passedTestingPhase";
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/Constants.java]: The name of the variable is too specific. Why should the `CreateOrUpdateAppStep` know about the existence of a testing phase? What we want to tell the step via this variable is whether it should skip the bindings or not.

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateOrUpdateAppStep.java [hunk 3]
```diff
@@ -245,6 +251,14 @@ public void handleApplicationAttributes() {
 
         @Override
         public void handleApplicationServices() throws FileStorageException {
+            Boolean passedTestingPhase = (Boolean) execution.getContext()
+                                                            .getVariable(Constants.VAR_PASSED_TESTING_PHASE);
+            if (processTypeParser.getProcessType(execution.getContext())
+                                 .equals(ProcessType.BLUE_GREEN_DEPLOY) &&
+                BooleanUtils.toBooleanDefaultIfNull(passedTestingPhase, false)) {
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateOrUpdateAppStep.java]: Why not handle this in a getter in `StepsUtil`. There we already have a generic method that allows you to provide a default value.

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateOrUpdateAppStep.java [hunk 4]
```diff
@@ -245,6 +251,14 @@ public void handleApplicationAttributes() {
 
         @Override
         public void handleApplicationServices() throws FileStorageException {
+            Boolean passedTestingPhase = (Boolean) execution.getContext()
+                                                            .getVariable(Constants.VAR_PASSED_TESTING_PHASE);
+            if (processTypeParser.getProcessType(execution.getContext())
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateOrUpdateAppStep.java]: Does it matter whether the deployment is "blue-green" or not? If the variable is set to `true` we should skip the bindings.

File: com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/CreateOrUpdateStepWithExistingAppTest.java [hunk 5]
```diff
@@ -3,26 +3,26 @@
...
 import org.cloudfoundry.client.lib.domain.CloudApplication.State;
 import org.flowable.engine.delegate.DelegateExecution;
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.ExpectedException;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-import org.junit.runners.Parameterized.Parameters;
+import org.junit.jupiter.api.Assertions;
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/CreateOrUpdateStepWithExistingAppTest.java]: Are you absolutely sure this works? `SyncFlowableStepTest` has JUnit4 annotations, while here you're using JUnit5. Check whether this test is not lying about being successful both in the IDE and in the console (`mvn clean install`). I remember that with some versions of JUnit4 and JUnit5, the integration did not work well. The test would pass successfully, even when it should've failed.

Comment (radito3) [reply com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/CreateOrUpdateStepWithExistingAppTest.java]: It works

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/Constants.java (hunk 2 above)
Comment (radito3) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/Constants.java]: Changed to "shouldSkipServiceRebinding"

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateOrUpdateAppStep.java (hunk 3 above)
Comment (radito3) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateOrUpdateAppStep.java]: Changed to `StepsUtil.getBoolean(context, varName, defaultValue)` and removed check for blue-green deploy process id

**Pull request merged**